src/Formats/FormatFactory.cpp

```diff
@@ -162,14 +162,12 @@ InputFormatPtr FormatFactory::getInput(
     if (settings.max_memory_usage_for_user && settings.min_chunk_bytes_for_parallel_parsing * settings.max_threads * 2 > settings.max_memory_usage_for_user)
         parallel_parsing = false;
 
-    if (parallel_parsing && name == "JSONEachRow")
+    if (parallel_parsing)
     {
-        /// FIXME ParallelParsingBlockInputStream doesn't support formats with non-trivial readPrefix() and readSuffix()
-
-        /// For JSONEachRow we can safely skip whitespace characters
-        skipWhitespaceIfAny(buf);
-        if (buf.eof() || *buf.position() == '[')
-            parallel_parsing = false; /// Disable it for JSONEachRow if data is in square brackets (see JSONEachRowRowInputFormat)
+        const auto & non_trivial_prefix_and_suffix_checker = getCreators(name).non_trivial_prefix_and_suffix_checker;
+        /// Disable parallel parsing for input formats with non-trivial readPrefix() and readSuffix().
```

Comment:
Can we just make it a bool false flag? The amount of crutches here is becoming unmanageable. Instead we should parse the prefix properly when parsing in parallel (this is a topic for a separate PR).

Reply:
Disabling it for JSONEachRow would technically be a performance regression, but maybe it'll give us a chance to implement proper parallel parsing already.
Added here https://github.com/ClickHouse/ClickHouse/pull/8958

Reply:
> Can we just make it a bool false flag?

If you are about non_trivial_prefix_and_suffix_checker then I guess no, because we need a function that checks the prefix  for a specific format.
>  but maybe it'll give us a chance to implement proper parallel parsing already

It would be great.